File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        
        int r=0, l= numbers.length -1;
        while (r<l){
            if(numbers[r] + numbers[l] == target){
                return new int [] {r+1,l+1};
            }
            else if (numbers[r] + numbers[l] >target){
                l --;
            }
            else if (numbers[r] + numbers[l] < target){
                r ++;
            }
        }
        return new int [] {0,0};
}
}

File: Data Structures & Algorithms/combination-target-sum/submission-1.cpp
class Solution {
public:
    vector<vector<int>> combinationSum(vector<int>& nums, int target) {
        vector<int> sub;
        vector<vector<int>> res;
        dfs(nums,target,sub,res,0);
        return res;
    }
    void dfs(vector<int> &nums , int target , vector<int> &sub , vector<vector<int>> &res , int i){
            if(target==0){
                res.push_back(sub);
                return ;
            }
            if(target<0 || i>=nums.size()){
                return ;
            }
        sub.push_back(nums[i]);
        dfs(nums,target-nums[i],sub,res,i);

        sub.pop_back();
        dfs(nums,target,sub,res,i+1);

    }
};
